cpp/include/cudf/strings/attributes.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf
+{
+namespace strings
+{
+
+/**
+ * @brief Returns an integer numeric column containing the length of each string in
+ * characters.
+ *
+ * The output column will have the same number of rows as the
+ * specified strings column. Each row value will be the number of
+ * characters in the corresponding string.
+ *
+ * Any null string will result in a null entry for that row in the output column.
+ *
+ * @throw cudf::logic_error if output_type is not an integer type.
+ *
+ * @param strings Strings instance for this operation.
+ * @param output_type The integer type used for creating the output column.
+ *        Default type is INT32.
+ * @param mr Resource for allocating device memory.
+ * @param stream Stream to use for any kernels in this function.
+ * @return New column with lengths for each string.
+ */
+std::unique_ptr<cudf::column> characters_counts( strings_column_view strings,
+                                                 data_type output_type = data_type{INT32},
+                                                 rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                                 cudaStream_t stream = 0);
```

Comment:
streams before mrs.

Reply:
Found this in the transition guide: https://github.com/rapidsai/cudf/blob/branch-0.11/cpp/docs/TRANSITIONGUIDE.md

> The recommended pattern for doing this is to make the definition of the external API invoke an internal API in the detail namespace. The internal detail API will have all the same parameters, plus a cudaStream_t parameter at the end defaulted to 0. The implementation should be wholly contained in the detail API definition and use only asynchronous versions of CUDA APIs with the defaulted stream parameter.

Reply:
And so it looks like I need to add detail surrogate functions for all my APIs.

Reply:
Even so, in most of the APIs that have been ported so far, `stream` is before `mr` in the parameter list. Is there a reason to reverse it here?

Reply:
I removed the parameter and placed it in a detail function as per the transition guide.